rdkit/Chem/Fraggle/FraggleSim.py

```diff
@@ -29,356 +29,285 @@
+    return True, m.GetNumAtoms()
+  else:
+    return False, 0
 
 
 def select_fragments(f_smi, ftype, hac):
-
-  result = ""
-  result_hcount = 0
   fragments = f_smi.split(".")
 
-  if (ftype == "acyclic"):
+  if ftype == FTYPE_ACYCLIC:
+    result = []
+    result_hcount = 0
     for f in fragments:
-      attachments = f.count("*")
-
-      #check if terminal fragment
-      if (attachments == 1):
+      # check if terminal fragment
+      if f.count("*") == 1:
         fMol = Chem.MolFromSmiles(f)
         fhac = fMol.GetNumAtoms()
 
-        #if the fragment is 2 atoms (or less - includes attachement) it is too small
-        #to be interesting. This check has the additional benefit
-        #of pulling out the relevant single cuts as it discards
-        #fragments where we only chop off a small part of the input cmpd
-        if (fhac > 3):
-          result = "%s.%s" % (result, f)
+        # if the fragment is 2 atoms (or less - includes attachement) it is too small
+        # to be interesting. This check has the additional benefit
+        # of pulling out the relevant single cuts as it discards
+        # fragments where we only chop off a small part of the input cmpd
+        if fhac > 3:
+          result.append(f)
           result_hcount = result_hcount + fhac
 
-    #needs to be greater than 60% of parent mol
-    if ((result != "") and (result_hcount > 0.6 * hac)):
-      #remove first character as it will always be "."
-      result = result[1:]
+    # needs to be greater than 60% of parent mol
+    if result and (result_hcount > 0.6 * hac):
+      result = '.'.join(result)
     else:
       result = None
+    return result
 
-  elif (ftype == "cyclic"):
+  elif ftype == FTYPE_CYCLIC:
+    # make sure it is 2 components
+    if len(fragments) != 2:
+      return None
     result = None
-    #make sure it is 2 components
-    if (len(fragments) == 2):
-      for f in fragments:
-        #check if a valid cut
-        valid, result_hcount = is_ring_cut_valid(f)
-
-        if (valid):
-          #needs to be greater 3 heavy atoms and greater than 40% of parent mol
-          if ((result_hcount > 3) and (result_hcount > 0.4 * hac)):
-            result = f
-
-  elif (ftype == "cyclic_and_acyclic"):
-    #print f_smi
-    result = ""
-
-    #need to find the fragments which are valid which means they must be:
+    for f in fragments:
+      # check if a valid cut
+      valid, result_hcount = is_ring_cut_valid(f)
+      if valid:
+        # needs to be greater 3 heavy atoms and greater than 40% of parent mol
+        if (result_hcount > 3) and (result_hcount > 0.4 * hac):
+          result = f
+    return result
+
+  elif (ftype == FTYPE_CYCLIC_ACYCLIC):
+    result = []
+
+    # need to find the fragments which are valid which means they must be:
     #  Terminal (one attachment point) or valid ring cut
     for f in fragments:
       attachments = f.count("*")
```

Comment:
Minor point, but it feels like this should be named nAttachments or attachmentsCount or something like that